### PR TITLE
Case id 213

### PR DESCRIPTION
**Pull request by MarkShleifer**

### CODE REVIEW

File: src/app/ansyn/ansyn.component.ts [hunk 1]
```diff
@@ -5,10 +5,11 @@ import { IStatusBarState, MapsLayout } from '@ansyn/status-bar/reducers/status-b
 import { Observable } from 'rxjs/Observable';
 import { ICasesState } from '@ansyn/menu-items/cases/reducers/cases.reducer';
 import { Case } from '@ansyn/menu-items/cases/models/case.model';
-import { cloneDeep } from 'lodash'
 import { UpdateCaseAction } from '@ansyn/menu-items/cases/actions/cases.actions';
-import * as _ from 'lodash';
 import "@ansyn/core/utils/clone-deep";
+import { ActivatedRoute, Params } from '@angular/router';
+import { LoadCaseAction, LoadDefaultCaseAction } from '../packages/menu-items/cases/actions/cases.actions';
+import * as _ from 'lodash';
```
Comment (MarkShleifer) [inline src/app/ansyn/ansyn.component.ts]: import isNil from lodash

File: src/app/packages/menu-items/cases/components/save-case/save-case.component.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    <span class="highlight"></span>
+    <span class="bar"></span>
+    <label>Name</label>
+
+    <div class="bottom-btns">
+      <button type="button" (click)="close()">Cancel</button>
+      <button type="submit">OK</button>
+    </div>
+  </form>
+</div>
```
Comment (MarkShleifer) [inline src/app/packages/menu-items/cases/components/save-case/save-case.component.html]: EOF

File: src/app/packages/menu-items/cases/models/map-state.model.ts [hunk 3]
```diff
@@ -9,4 +9,4 @@ export type MapState = {
 };
 
 // TODO: Need to get the real map Type from store
-export const defaultMapType = "openLayersMap";
+export const defaultMapType = "openLayersMap";
```
Comment (MarkShleifer) [inline src/app/packages/menu-items/cases/models/map-state.model.ts]: EOF

File: src/app/packages/menu/menu/menu.component.ts [hunk 4]
```diff
@@ -54,12 +54,22 @@ export class MenuComponent {
...
+
+	constructor(public componentFactoryResolver: ComponentFactoryResolver,
+		private store: Store<IMenuState>) {
+
+		this.menu_items$.subscribe((menu_items: MenuItem[]) => {
+			this.menu_items = menu_items;
+		});
+
+		this.selected_menu_item_index$
+			.debug()
```
Comment (MarkShleifer) [inline src/app/packages/menu/menu/menu.component.ts]: remove debug

File: src/app/packages/menu/menu/menu.component.ts [hunk 5]
```diff
@@ -121,19 +133,23 @@ export class MenuComponent {
...
 			this.componentChanges();
-			if(this.itemSelected()){
+			if (this.itemSelected()) {
 				this.expand = true;
 			}
 		}
 	}
+
+	ngAfterViewInit() {
+		this.componentChanges();
```
Comment (MarkShleifer) [inline src/app/packages/menu/menu/menu.component.ts]: check performance

**Comment by osadan** [comment]
Don't forget to get the new layers.json

**Pull request merged**